fsck.c

```diff
@@ -1323,6 +1337,62 @@ int fsck_finish(struct fsck_options *options)
 	return ret;
 }
 
+__attribute__((format (printf, 4, 5)))
+int fsck_refs_report(struct fsck_refs_options *o,
+		     const char *name,
+		     enum fsck_msg_id msg_id,
+		     const char *fmt, ...)
+{
+	va_list ap;
+	struct strbuf sb = STRBUF_INIT;
+	enum fsck_msg_type msg_type =
+		fsck_msg_type(msg_id, (struct fsck_options*)o);
+	int ret = 0;
+
+	if (msg_type == FSCK_IGNORE)
+		return 0;
+
+	if (msg_type == FSCK_FATAL)
+		msg_type = FSCK_ERROR;
+	else if (msg_type == FSCK_INFO)
+		msg_type = FSCK_WARN;
+
+	prepare_msg_ids();
+	strbuf_addf(&sb, "%s: ", msg_id_info[msg_id].camelcased);
+
+	va_start(ap, fmt);
+	strbuf_vaddf(&sb, fmt, ap);
+	ret = o->error_func(o, name, msg_type, msg_id, sb.buf);
+	strbuf_release(&sb);
+	va_end(ap);
+
+	return ret;
+}
+
+int git_fsck_refs_config(const char *var, const char *value,
```

Comment:
Why not use `git_fsck_config`, the only thing it does differently is setting the skiplist, which we don't have to consider anyways.

Reply:
`git_fsck_config` was designed to set up the `fsck_options`. I have created a new structure `fsck_refs_options`, so I think we should create a new function to set up the `fsck_refs_options`.

BTY, I want to explain why I designed this here. The git-fsck(1) is highly relevant with the object database. I don't want to add some new fields into `fsck_options` due to the following reasons:

1. The fields in `fsck_options` are all related to objects except `fsck_msg_type` and `strict`. Adding some ref-related fields into `fsck_options` will break the semantics.
2. We may later change `fsck_options` to `fsck_objs_options`.

For refs and objects, the configuration should be different, so I create a new function. Actually, they DO share some common operations. I think we should handle this later. As Patrick said:

> My long-term vision here would actually rip the ODB-related stuff out of git-fsck(1) and into a new git objects verify command.

Actually, it may be perfect that I may abstract some interfaces here, however, it would be too complicated, many functions use `fsck_options *` as the parameter. I think we may do this later.

I will mention above idea into commit message to make the intention more clear.

Reply:
After discussing, we decided to just let the design unchanged.

Will send the patch to the community to see whether this is OK!